Modules/Core/Common/src/itkPoolMultiThreader.cxx

```diff
@@ -30,12 +30,60 @@
 #include "itkProcessObject.h"
 #include "itkImageSourceCommon.h"
 #include <algorithm>
+#include <exception>
 #include <iostream>
 #include <string>
-#include <algorithm>
 
 namespace itk
 {
+namespace
+{
+class ExceptionHandler
+{
+public:
+  template <typename TFunction>
+  explicit ExceptionHandler(const TFunction & function)
+  {
+    try
+    {
+      function();
+    }
+    catch (...)
```

Comment:
Honestly I believe an exception should always be derived from `std::exception`, as a general C++ coding conversion, so it should be sufficient to do `catch (const std::exception&)`, instead of `catch (...)`. What do you think?

Reply:
This code needs to be bullet proof. If an exception is not caught it can lead to segmentation faults, abandoned threads, and other bad things.  It wraps the execution of user's threaded code. We don't have any control of what some one may running threaded. We need to be able to handle anything they throw at us ( pun intended ).

Reply:
That will hold water until someone comes along who writes `throw "Error HEREEEEEEE";` and then comes crying to the forum because his program is crashing without explanation.

People don't always follow conventions, so we should handle all exceptions.

Reply:
OK, no problem, I will leave the `catch(...)` in there, if this is the way you want it.  It's just that https://www.boost.org/community/error_handling.html said so, many years ago:

> Avoid catch(...) when possible

Reply:
This handles exceptions both from our and other people's code. We can't expect everyone to be nice. Avoiding `catch(...)` here is quite detrimental.